### PR TITLE
Reproduce and fix #1272

### LINKED ISSUE #1272: java.lang.IllegalArgumentException: Error: missing bug code for keyJUA
When running spotbugs through Gradle I get the following exception:

```
The following errors occurred during analysis:
  Could not generate HTML output
    java.lang.IllegalArgumentException: Error: missing bug code for keyJUA
      At edu.umd.cs.findbugs.DetectorFactoryCollection.getBugCode(DetectorFactoryCollection.java:509)
      At edu.umd.cs.findbugs.BugPattern.getBugCode(BugPattern.java:121)
      At edu.umd.cs.findbugs.BugInstance.getCWEid(BugInstance.java:1961)
      At edu.umd.cs.findbugs.BugInstance.writeXML(BugInstance.java:1989)
      At edu.umd.cs.findbugs.SortedBugCollection.writeXML(SortedBugCollection.java:554)
      At edu.umd.cs.findbugs.SortedBugCollection.toDocument(SortedBugCollection.java:443)
      At edu.umd.cs.findbugs.HTMLBugReporter.finish(HTMLBugReporter.java:52)
      At edu.umd.cs.findbugs.DelegatingBugReporter.finish(DelegatingBugReporter.java:89)
      At edu.umd.cs.findbugs.DelegatingBugReporter.finish(DelegatingBugReporter.java:89)
      At edu.umd.cs.findbugs.FindBugs2.analyzeApplication(FindBugs2.java:1165)
      At edu.umd.cs.findbugs.FindBugs2.execute(FindBugs2.java:309)
      At com.github.spotbugs.snom.internal.SpotBugsRunnerForWorker$SpotBugsExecutor.execute(SpotBugsRunnerForWorker.java:97)
      At org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:63)
      At org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:49)
      At org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:43)
      At org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:97)
      At org.gradle.workers.internal.AbstractClassLoaderWorker.executeInClassLoader(AbstractClassLoaderWorker.java:43)
      At org.gradle.workers.internal.IsolatedClassloaderWorker.run(IsolatedClassloaderWorker.java:49)
      At org.gradle.workers.internal.IsolatedClassloaderWorker.run(IsolatedClassloaderWorker.java:30)
      At org.gradle.workers.internal.WorkerDaemonServer.run(WorkerDaemonServer.java:85)
      At org.gradle.workers.internal.WorkerDaemonServer.run(WorkerDaemonServer.java:55)
      At org.gradle.process.internal.worker.request.WorkerAction$1.call(WorkerAction.java:138)
      At org.gradle.process.internal.worker.child.WorkerLogEventListener.withWorkerLoggingProtocol(WorkerLogEventListener.java:41)
      At org.gradle.process.internal.worker.request.WorkerAction.run(WorkerAction.java:135)
      At sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
      At sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
      At sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
      At java.lang.reflect.Method.invoke(Method.java:498)
      At org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
      At org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
      At org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConne
ction.java:182)
      At org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConne
ction.java:164)
      At org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
      At org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
      At org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
      At java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
      At java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
      At org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
      At java.lang.Thread.run(Thread.java:748)
```

SpotBugs: 4.1.2
Gradle Plugin: 4.5.0
Gradle: 6.6

This does not happen in 4.1.1!

### PR DESCRIPTION
**Pull request by KengoTODA**
This PR reproduces and fixes #1272.

- [x] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,9 @@ This is the changelog for SpotBugs. This follows [Keep a Changelog v1.0.0](http:
 Currently the versioning policy of this project follows [Semantic Versioning v2.0.0](http://semver.org/spec/v2.0.0.html).
 
 ## Unreleased - 2020-??-??
+### Fixed
+* `IllegalArgumentException` during XML report generation ([#1272](https://github.com/spotbugs/spotbugs/issues/1272))
+
 ### Changed
 * Bump jaxen from 1.1.6 to 1.2.0 supporting Java 11 compilation ([#1316](https://github.com/spotbugs/spotbugs/issues/1316))
 * Fixed error dialog on cancelling SpotBugs job in Eclipse ([#1314](https://github.com/spotbugs/spotbugs/issues/1314))
```
Comment (KengoTODA) [inline CHANGELOG.md:13]: note: I'll move these two `fixed` entries to `### Fixed` part in another PR.

File: spotbugs/etc/messages_fr.xml [hunk 2]
```diff
@@ -3501,4 +3501,5 @@ double value2 =  x / (double) y;
  <BugCode abbrev="QBA">Assignation booléenne douteuse</BugCode>
  <BugCode abbrev="VR">Problème de compatibilité entre les versions</BugCode>
  <BugCode abbrev="DP">Utiliser doPrivileged</BugCode>
+ <BugCode abbrev="JUA">Problems in JUnit Assertions</BugCode>
```
Comment (jscancella) [inline spotbugs/etc/messages_fr.xml:3504]: Did you mean to put an english message in the french messages bundle?

**Review by jscancella — COMMENTED** [review]
LGTM, just need to have someone who understands french change the message for the french bundle.

**Review by jscancella — DISMISSED** [review]

File: spotbugs/etc/messages_fr.xml (hunk 2 above)
Comment (iloveeclipse) [reply spotbugs/etc/messages_fr.xml:3504]: `Problème in JUnit Assertions` ?

Comment (KengoTODA) [reply spotbugs/etc/messages_fr.xml:3504]: [Google translation](https://translate.google.com/#view=home&op=translate&sl=en&tl=fr&text=Problems%20in%20JUnit%20Assertions) and [Deepl](https://www.deepl.com/translator#en/fr/Problems%20in%20JUnit%20Assertions) gave different answers for this sentence, so I will remove this change in this branch. It should work even in `fr` users (SpotBugs uses the English sentence instead).

**Review by ThrawnCA — APPROVED** [review]

**Review by wreulicke — APPROVED** [review]

**Pull request merged**